0000-miner-update.md

```diff
@@ -0,0 +1,199 @@
+Helium will solely continue to update using our existing nextgate/nexthaul
+client-server mechanism.
+
+# Motivation
+[motivation]: #motivation
+
+- Why are we doing this?
+- What use cases does it support?
+- What problems does it solve?
+- What is the expected outcome?
+
+Allowing Hotspot owners to build and distribute their own custom miner software
+encourages diversity across the Helium Network. Delivering different miner
+updates from different points of origin allows the network to grow in a truly
+decentralized fashion, eventually moving control of how the network develops
+away from any single entity and into the hands of the network stakeholders
+themselves.
+
+# Stakeholders
+[stakeholders]: #stakeholders
+
+* Who is affected by this HIP?
+
+Blockchain Engineering, Embedded Engineering
+
+* How are we soliciting feedback on this HIP from these stakeholders?
+
+Feedback will be gathered by sharing this HIP.
+
+# Detailed Explanation
+[detailed-explanation]: #detailed-explanation
+
+Helium's private nextgate build system builds Hotspot firmware, signs it and
+uploads it to a public AWS S3 bucket for downloading. The nexthaul OTA server
+provides metadata about where to find and how to verify available Hotspot
+firmware releases. The Helium Blockchain Engineering team updates this release
+information inside the helium/coop repo and pushes changes to Heroku for
+deployment to the existing fleet of Hotspots. Hotspots use a self-signed public
+SSL certificate included in their firmware to establish that the ota.helium.com
+server on the other end is in fact Helium's and not some bad actor's.
+
+The nextgate repo currently builds miner as a separate software package that is
+then installed to a target root file system for inclusion in the Hotspot's
+embedded Linux image. The entire system image is signed by Helium but the
+individual miner software artifacts currently are not. The firmware image is
+signed so that consumer Hotspots can verify that an over-the-air update
+originated from Helium before committing it to flash storage and rebooting.
+
+The monolithic firmware updates that nextgate/nexthaul performs are too
+dangerous for Hotspot owners to administer themselves. The chances of an end
+user deploying a homemade firmware update that accidentally bricks their
+Hotspots is simply too high. What we want is a way for Hotspot owners to only
+upgrade the miner portion of their devices' firmware so that their Hotspots
+remain online and continue to receive hardware-specific updates (Linux kernel,
+gateway-config, LoRa packet_forwarder) only from Helium.
+
+The simplest solution is to install miner and all its dependencies (RocksDB,
+libsodium, ebus) as an image overlay on top of a Hotspot's root file system.
+This extraction step would happen at boot time once Linux has started and the
+immutable root file system has been loaded into memory. The end result is
```

Comment:
So the user needs to push a whole tarball out to their hotspot somehow? If there was a UX to indicate the source of the update, we would need to make that service overwrite some SD card contents to store the tarball? or I suppose you could store it in `var`?

Reply:
> So the user needs to push a whole tarball out to their hotspot somehow?

Yes.  They would push their own miner updates via a minerhaul GitHub fork that they deploy to Heroku.

> We would need to make that service overwrite some SD card contents to store the tarball?

Correct.  We could do an atomic move like we do for the embedded Linux image updates in a `MINER_PACKAGE_DIR` inside `var`.